compiler/infra/OMRCfg.cpp

```diff
@@ -228,7 +259,70 @@ OMR::CFG::addExceptionEdge(
       }
    }
 
+void
+OMR::CFG::addExceptionEdge(
+      TR::CFGNode *f,
+      TR::CFGNode *t)
+   {
+   if (comp()->getOption(TR_TraceAddAndRemoveEdge))
+      {
+      traceMsg(comp(),"\nAdding exception edge %d-->%d:\n", f->getNumber(), t->getNumber());
+      }
+
+   TR_ASSERT(!f->hasSuccessor(t), "adding an exception edge when there's already a non exception edge");
+   TR::Block * newCatchBlock = toBlock(t);
+   for (auto e = f->getExceptionSuccessors().begin(); e != f->getExceptionSuccessors().end(); ++e)
+      {
+      TR::Block * existingCatchBlock = toBlock((*e)->getTo());
+      if (newCatchBlock == existingCatchBlock) return;
+
+      // OSR exception edges are special and we do not want any 'optimization' done to them
+      // from the following special checks
+      if (newCatchBlock->isOSRCatchBlock() || existingCatchBlock->isOSRCatchBlock()) continue;
+
+      // If the existing catch block is going to be considered first and it catches everything that
+      // the new catch block does then the new catch block isn't reaching from the 'f' block
+      //
+      // Catch block 'A' is considered before catch block 'B' if 'A' is from a greater inline depth
+      // or 'A' and 'B' are from the same inline depth and 'A' handler index is less than 'B's.
+      //
+      int32_t existingDepth = existingCatchBlock->getInlineDepth();
+      int32_t newDepth = newCatchBlock->getInlineDepth();
+      if (existingDepth < newDepth ||
+          (existingDepth == newDepth && existingCatchBlock->getHandlerIndex() > newCatchBlock->getHandlerIndex()))
+         continue;
+
+      // The existing catch block is going to be considered first.  Don't add an edge to the new catch
+      // block if the existing one catches everything that the new one catches.
+      //
+      /////void * newEC = newCatchBlock->getExceptionClass(), * existingEC = existingCatchBlock->getExceptionClass();
```

Comment:
Does this commented out code serve any purpose?

Reply:
The commented out code is also found in the deprecated API. This is just a literal copy paste. By this I mean: since the code was commented out originally by someone else, and it has been commented since the [initial contribution](https://github.com/eclipse/omr/blame/master/compiler/infra/OMRCfg.cpp#L202), I don't believe it serves any purpose. I can delete the comments if you want.

Regarding the code duplication, I don't think it should be an issue. As soon as this change (and one in OpenJ9) are pulled, then I can come back and delete the previous implementation.

Reply:
I see. I'm fine with keeping the comments if they are just copy-pasted from old code. I'll leave it up to you to decide if you want to remove them or open an issue about it instead (it would be a good beginner item).